domain/consensus/model/pow/pow.go

```diff
@@ -0,0 +1,55 @@
+package pow
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/hashes"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/serialization"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+// CheckProofOfWork check's if the block has a valid PoW according to its Bits field
+// it does not check if the difficulty itself is valid or less than the maximum for the appropriate network
+func CheckProofOfWork(header *externalapi.DomainBlockHeader, optionalTarget *big.Int) bool {
+	if optionalTarget == nil {
```

Comment:
This is not very go-y to have optionalTarget as parameter.

Instead better do something like (Names were not thought through, feel free to alter):
```
CheckProofOfWorkWithTarget(header, target){
 // Do whatever
}

CheckProofOfWorkAccordingToBits(header){
    CheckProofOfWorkWithTarget(header, util.CompactToBig(header.Bits))
}

```